HopsanCore/include/ComponentUtilities/EquationSystemSolver.h

```diff
@@ -100,6 +110,30 @@ class HOPSANCORE_DLLAPI NumericalIntegrationSolver
 };
 
 
+class HOPSANCORE_DLLAPI KinsolSolver
+{
+public:
+    enum SolverTypeEnum {NewtonIteration=0,FixedPointIteration=1};
+
+    KinsolSolver(Component *pParentComponent, double tol, int n, SolverTypeEnum solverType);
+    ~KinsolSolver();
+    void solve();
+    double getState(int i);
+    void setState(int i, double value);
+    void setTolerance(double value);
+private:
+    Component *mpParentComponent;
+    void *mem;
+#ifdef USESUNDIALS
+    N_Vector y;
```

Comment:
I dont think that sundials types should be exposed by our headers, they should be kept internal in our cpp files.
I recommend using "pimpl" solution.
Also having conditional layout of this class is a problem if you build the core with one setting (with or without) than try to link that version into an application where the macro is different for some reason. Then the class layout will mismatch, and we get memory corruption. 

In this case it seems especially unnecessary since the members are all private anyway.
Then we should not need to include the sundils.pri file everywhere as well.

Reply:
Good suggestion, that was actually the solution I was looking for. Now everything became much nicer.